Kernel/FileSystem/MountFile.cpp

```diff
@@ -0,0 +1,112 @@
+    // to ensure this never happens.
+    VERIFY(!(flags & MS_BIND));
+    VERIFY(!(flags & MS_REMOUNT));
+    auto mount_specific_data_buffer = TRY(KBuffer::try_create_with_size("Mount Specific Data"sv, 4096, Memory::Region::Access::ReadWrite, AllocationStrategy::AllocateNow));
+    return TRY(adopt_nonnull_lock_ref_or_enomem(new (nothrow) MountFile(file_system_initializer, flags, move(mount_specific_data_buffer))));
+}
+
+MountFile::MountFile(FileSystemInitializer const& file_system_initializer, int flags, NonnullOwnPtr<KBuffer> mount_specific_data)
+    : m_flags(flags)
+    , m_file_system_initializer(file_system_initializer)
+{
+    m_mount_specific_data.with_exclusive([&](auto& our_mount_specific_data) {
+        our_mount_specific_data = move(mount_specific_data);
+        memset(our_mount_specific_data->data(), 0, our_mount_specific_data->size());
+    });
+}
+
+MountFile::~MountFile() = default;
+
+ErrorOr<void> MountFile::ioctl(OpenFileDescription&, unsigned request, Userspace<void*> arg)
+{
+    return m_mount_specific_data.with_exclusive([&](auto& our_mount_specific_data) -> ErrorOr<void> {
+        switch (request) {
+        case MOUNT_IOCTL_SET_MOUNT_SPECIFIC_FLAG: {
+            auto user_mount_specific_data = static_ptr_cast<mount_specific_flag const*>(arg);
+            auto mount_specific_data = TRY(copy_typed_from_user(user_mount_specific_data));
+            if ((mount_specific_data.value_type == mount_specific_flag::ValueType::SignedInteger || mount_specific_data.value_type == mount_specific_flag::ValueType::UnsignedInteger) && mount_specific_data.value_length != 8)
+                return EDOM;
+            if (mount_specific_data.key_string_length > 64)
+                return ENAMETOOLONG;
+            if (mount_specific_data.value_type != mount_specific_flag::ValueType::None && mount_specific_data.value_length == 0)
+                return EINVAL;
+            if (mount_specific_data.value_type != mount_specific_flag::ValueType::None && mount_specific_data.value_addr == nullptr)
+                return EFAULT;
+            if (mount_specific_data.value_type != mount_specific_flag::ValueType::ASCIIString && mount_specific_data.value_length > 64)
+                return E2BIG;
+            if (mount_specific_data.value_type == mount_specific_flag::ValueType::ASCIIString && mount_specific_data.value_length > 1024)
+                return E2BIG;
+            // NOTE: We enforce that the passed argument will be either i64 or u64, so it will always be
+            // exactly 8 bytes. We do that to simplify handling of integers as well as to ensure ABI correctness
+            // in all possible cases.
+            auto key_string = TRY(try_copy_kstring_from_user({ reinterpret_cast<char const*>(mount_specific_data.key_string_addr), static_cast<size_t>(mount_specific_data.key_string_length) }));
+            switch (mount_specific_data.value_type) {
+            // NOTE: This is actually considered as simply boolean flag.
```

Comment:
no NOTE

Reply:
I already used NOTE a lot in the past (and other people did that too) and I see no problems with this pattern. This helps differentiate this from FIXME notes, for example.

Reply:
After the discussion in the discord server I think it's fine to let it be this way.